File: exercicios_ruby/exercicios/exercicio2.rb
# Os triângulos podem ser classificados em 3 tipos quanto ao tamanho de seus lados:
# Equilátero: Os três lados são iguais. Isósceles: Dois lados iguais. Escaleno: Todos os lados são diferentes. Crie uma função que recebe os comprimentos dos três lados de um triângulo e retorne sua classificação quanto ao tamanho de seus lados. (Neste exemplo deve-se abstrair as condições matemáticas de existência de um triângulo).

def classifica_triangulo(a, b, c)
  if triangulo?(a, b, c,)
    if a == b && b == c
      puts 'Equilátero'
    elsif a != b && b != c
      puts 'Escaleno'
    else
      puts 'Isósceles'
    end
  else
    puts 'As retas não formam um triangulo!'
  end

end

def triangulo?(a, b, c)
  if a + b >  c && b + c > a && a + c > b
    true
  else
    false
  end
end

classifica_triangulo(5, 5, 5)

File: exercicios_python/exercicios/exercicio2.py
#  Os triângulos podem ser classificados em 3 tipos quanto ao tamanho de seus lados:  
# Equilátero: Os três lados são iguais. Isósceles:  Dois lados iguais. Escaleno: Todos os lados são diferentes.  
# Crie uma função que recebe os comprimentos dos três lados de um triângulo e retorne sua classificação quanto 
# ao tamanho de seus lados. (Neste exemplo deve-se abstrair as condições matemáticas de existência de um 
# triângulo).

def classifica_triangulo(a, b, c):

  if is_triangulo(a, b, c) == True:
    if a == b and a == c:
      print("Equilátero")
    elif a != b and a != c:
      print("Isósceles")
    else:
      print("Escaleno")
  else:
    print("A soma dos lados não pode formar um Triangulo")




def is_triangulo(a, b, c):
  if a + b > c and a + c > b and b + c > a:
    return  True
  else:
    return False



classifica_triangulo(5, 4, 3)